jobs/somalier.py

```diff
@@ -0,0 +1,364 @@
+"""
+Adding jobs for fingerprinting and pedigree checks. Mostly using Somalier.
+"""
+import logging
+from os.path import basename
+
+import pandas as pd
+from hailtop.batch import Batch, Resource
+from hailtop.batch.job import Job
+
+from cpg_utils import Path, to_path
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import (
+    image_path,
+    reference_path,
+    fasta_res_group,
+    copy_common_env,
+    command,
+)
+from cpg_utils.workflows.resources import STANDARD
+from cpg_utils.workflows.slack import slack_message_cmd
+from cpg_utils.workflows.targets import Dataset, Sample
+from cpg_utils.workflows.filetypes import CramPath, GvcfPath
+from cpg_utils.workflows.utils import can_reuse, rich_sample_id_seds
+
+from .python_scripts import check_pedigree
+
+
+# We exclude contaminated samples from relatedness checks
```

Comment:
Is that because contaminated samples would show up as related to a number of other samples? Would be great to expand the comment a bit.

Should this be called `MAX_FREEMIX` instead, as it's an upper bound?

Reply:
Yeah, a contaminated sample can negatively affect relatedness estimation results for the entire dataset, so we need to make sure we don't pass those to `somalier relate`.

 I'll expand that in a comment!